report_broken_clients.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"strings"
+
+	"github.com/groob/mackit/cfpref"
+)
+
+func serialNumber() string {
+	out, err := exec.Command("/usr/sbin/ioreg", "-l").Output()
+	if err != nil {
+		return ""
+	}
+	for _, l := range strings.Split(string(out), "\n") {
+		if strings.Contains(l, "IOPlatformSerialNumber") {
+			s := strings.Split(l, " ")
+			t := strings.Replace(s[len(s)-1], "\"", "", -1)
+			return t
+		}
+	}
+	return ""
+}
+
+func main() {
+	serial_number := serialNumber()
+	if serial_number == "" {
```

Comment:
you're checking for "" anyway and calling panic. what I would do instead is have `serialNumber() return `(string, error). that lets you check for `err != nil` in main and quit. 

at the bottom of the function I'd add `return "", errors.New("could not retrieve serial number")`